src/main/java/org/json/JSONArray.java

```diff
@@ -144,6 +144,42 @@ public JSONArray(JSONTokener x, JSONParserConfiguration jsonParserConfiguration)
                 }
             }
         }
+
+        if (jsonParserConfiguration.isStrictMode()) {
+            validateInput(x);
+        }
+    }
+
+    /**
+     * Checks if Array adheres to strict mode guidelines, if not, throws JSONException providing back the input in the
+     * error message.
+     *
+     * @param x tokener used to examine input.
+     * @throws JSONException if input is not compliant with strict mode guidelines;
+     */
+    private void validateInput(JSONTokener x) {
+        char cursor = x.getPrevious();
+
+        boolean isEndOfArray = cursor == ']';
+        boolean nextCharacterIsNotEoF = x.nextClean() != 0;
+
+        if (isEndOfArray && nextCharacterIsNotEoF) {
+            String completeInput = collectCompleteInput(x);
```

Comment:
It's probably not a good idea to collect complete input, in the case of a large JSONArray. Just the char value, index, and line number should be sufficient.

Reply:
Will do.